pkg/rkt-stage1/0007-Set-up-environment-for-the-stage2-container.patch

```diff
@@ -0,0 +1,45 @@
+From b51ab95f355b059664ff417c14f1cf44deb72683 Mon Sep 17 00:00:00 2001
+From: bharani-zededa <bharani@zededa.com>
+Date: Wed, 13 Nov 2019 11:08:55 -0800
+Subject: [PATCH] Set up environment for the stage2(container)
+
+---
+ files/run          | 9 +++++++++
+ kernel/init-initrd | 3 +++
+ 2 files changed, 12 insertions(+)
+
+diff --git a/files/run b/files/run
+index fbb3a71..482c0ef 100755
+--- a/files/run
++++ b/files/run
+@@ -94,6 +94,15 @@ else
+ fi    
+ 
+ echo $cmdline > $mountpoint/cmdline
++export IFS=$'\n'
++stage2manifest=$mountpoint/manifest
++stage2env=$mountpoint/environment
++for row in `cat $stage2manifest | jq -c .app.environment[]`; do
++  KEY=$(echo $row | jq -r .name)
++  VALUE=$(echo $row | jq -r .value)
++  echo export $KEY=\'$VALUE\' >> $stage2env
++done
++
+ xl create $gargs $outconfig $STAGE1_XL_OPTS
+ domname=$(sed -ne '/^name *=/s/^.*"\(.*\)"/\1/p' < "$outconfig")
+ domid=`xl list | grep "$domname" | awk '{print$2}'`
+diff --git a/kernel/init-initrd b/kernel/init-initrd
+index eb2716d..2a32add 100755
+--- a/kernel/init-initrd
++++ b/kernel/init-initrd
+@@ -76,4 +76,7 @@ then
+ fi
+ cmd=`cat /mnt/cmdline`
+ echo "Executing $cmd"
++echo "Inheriting environment"
++source /mnt/environment
```

Comment:
Can we print the Environment somewhere for debugging purposes?

Reply:
"export -p" prints the environment variables that are being set to the console. I added that for debugging. On thinking more, I will be removing it - the environment variables can have secrets credentials and such. So, it is not a good idea to print/log those.